config/unicorn.conf.rb

```diff
@@ -273,23 +234,6 @@ def check_email_sync_heartbeat
 
         Demon::EmailSync.restart if restart
       end
-
-      def master_sleep(sec)
-        sidekiqs = ENV["UNICORN_SIDEKIQS"].to_i
-        if sidekiqs > 0
-          Demon::Sidekiq.ensure_running
-          check_sidekiq_heartbeat
-        end
-
-        if ENV["DISCOURSE_ENABLE_EMAIL_SYNC_DEMON"] == "true"
-          Demon::EmailSync.ensure_running
-          check_email_sync_heartbeat
-        end
-
-        DiscoursePluginRegistry.demon_processes.each { |demon_class| demon_class.ensure_running }
-
-        master_sleep_orig(sec)
-      end
```

Comment:
I cannot determine why we need to be monkey patching unicorn here just to ensure that the child processes are still running. I also cannot think of a good reason why stuff is done this way. Instead, I removed the monkey patch and [introduced a long running thread](https://github.com/discourse/discourse/pull/30198/files#diff-a87fefdca78fa99e4442a097acfabea05ea813c56d91b26272dbd4fc04bf53c3R124-R147) which wakes up every 60 seconds to run checks instead.